Adding DLS with write permission recommendation #1273 (#7668)
opensearch-project/documentation-website: _security/access-control/document-level-security.md

OpenSearch combines all DLS queries with the logical `OR` operator. However, when a role that uses DLS is combined with another security role that doesn't use DLS, the query results are filtered to display only documents matching the DLS from the first role. This filter rule also applies to roles that do not grant read documents.

### DLS and write permissions

> It is recommended to always ensure that the user which has DLS configured roles does not have write permissions. If write permissions are added, the user will be able to index documents which they will not be able to retrieve due to DLS filtering.

Reviewer: ```suggestion
Make sure that a user that has DLS-configured roles does not have write permissions. If write permissions are added, the user will be able to index documents which they will not be able to retrieve due to DLS filtering.
```
